File: DynamicProgramming/392_IsSubsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        if len(t) == 0:
            return False
        c_idx = 0 
        curr = s[c_idx]
        for i in range(0, len(t)):
            if curr == t[i]:
                c_idx += 1
                if c_idx == len(s):
                    return True
                curr = s[c_idx]
        return False
        

File: DynamicProgramming/392_IsSubsequence.java
public class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0) {
            return true;
        }

        if (t.length() == 0) {
            return false;
        }

        int c_idx = 0;
        char curr = s.charAt(c_idx);
        for (int i=0; i<t.length(); i++) {
            if (curr == t.charAt(i)) {
                if (++c_idx == s.length()) {
                    return true;
                }
                curr = s.charAt(c_idx);
            }
        }
        return false;
    }
}